ivy/__init__.py

```diff
@@ -59,6 +63,22 @@ def __new__(cls, dtype_str):
         return str.__new__(cls, dtype_str)
 
 
+class Shape(tuple):
+    def __new__(cls, shape_tup):
+        assert isinstance(
+            shape_tup, (int, list, tuple, ivy.current_backend().NativeShape)
```

Comment:
you can replace `ivy.current_backend().NativeShape` with `ivy.NativeShape`, if a backend is currently set, it will have overwritten `ivy.NativeShape`